src/admin/resolvers/queries/ApprovedItem.ts

```diff
@@ -16,8 +19,16 @@ import {
 export async function getApprovedItems(
   parent,
   args,
-  { db }
+  context: IContext
 ): Promise<Connection<ApprovedItem>> {
+  //check if the user does not have the permissions to access this query
+  if (
+    !context.authenticatedUser.hasReadOnly &&
+    !context.authenticatedUser.canWriteToCorpus()
```

Comment:
This can also be written as 
`!(
    context.authenticatedUser.hasReadOnly ||
    context.authenticatedUser.canWriteToCorpus()
    )`
But I prefer it with the `&&`

Reply:
De Morgan's law :)